### PR TITLE
feat(datetime-picker): support spinner mode

### PR DESCRIPTION
**Pull request by c-clasing**
## Pull request checklist

Please check if your PR fulfills the following requirements:

- [x] The changes have been tested successfully.
- [x] A changeset has been created (`npm run changeset`).

### CODE REVIEW

**Comment by robingenz** [comment]
Thank you! I will test and review your PR as soon as possible.

**Comment by robingenz** [comment]
@c-clasing The `spinner` theme should also support the `time` mode.

**Comment by robingenz** [comment]
@c-clasing Thanks for the PR! This looks good. I am just wondering if `themes` is the right place for this configuration or if it is better to add a new parameter. I'll give it some thought this weekend.

**Comment by c-clasing** [comment]
> @c-clasing Thanks for the PR! This looks good. I am just wondering if `themes` is the right place for this configuration or if it is better to add a new parameter. I'll give it some thought this weekend.

Hello, are there any news on this? We are currently waiting for this and would prefer not using an local build in the project. 
If there is a need to change anything i'd be happy to do it.

**Comment by robingenz** [comment]
@c-clasing Sorry, I am very busy at the moment. I'll try to review it over the weekend so we can merge it next week.

**Comment by robingenz** [comment]
@c-clasing I finally found time to take a closer look at this.
Instead of modifying the `theme` option, I want to introduce two new options `androidTimePickerMode` and `androidDatePickerMode`:

```ts
export interface PresentOptions {
  ...
  /**
   * Whether to use the spinner or clock mode for the time picker on Android.
   * This value overwrites the `androidTimePickerMode` configuration value.
   *
   * Only available on Android.
   *
   * @since 5.1.0
   */
  androidTimePickerMode?: 'clock' | 'spinner';
  /**
   * Whether to use the calendar or spinner mode for the date picker on Android.
   * This value overwrites the `androidDatePickerMode` configuration value.
   *
   * Only available on Android.
   *
   * @since 5.1.0
   */
  androidDatePickerMode?: 'calendar' | 'spinner';
```

Please also extend the `PluginsConfig` interface:

```ts
  export interface PluginsConfig {
    /**
     * These configuration values are available:
     */
    DatetimePicker?: {
      ...
	  /**
	   * Whether to use the spinner or clock mode for the time picker on Android.
	   *
	   * Only available on Android.
	   *
	   * @since 5.1.0
	   * @default 'clock'
	   */
	  androidTimePickerMode?: 'clock' | 'spinner';
	  /**
	   * Whether to use the calendar or spinner mode for the date picker on Android.
	   *
	   * Only available on Android.
	   *
	   * @since 5.1.0
	   * @default 'calendar'
	   */
	  androidDatePickerMode?: 'calendar' | 'spinner';
    };
  }
```

After that, we can merge it.

File: packages/datetime-picker/src/web.ts
Comment (robingenz) [inline packages/datetime-picker/src/web.ts]: Please revert this change.

**Review by robingenz — CHANGES_REQUESTED** [review]

File: packages/datetime-picker/src/web.ts
Comment (c-clasing) [reply packages/datetime-picker/src/web.ts]: Okay, i just tried to fix the linter errors^^

**Comment by robingenz** [comment]
I just published the dev version:

```
npm i @capawesome-team/capacitor-datetime-picker@5.0.0-dev.da70913.1692263211
```

**Comment by Crylion** [comment]
> @capawesome-team/capacitor-datetime-picker@5.0.0-dev.da70913.1692263211

@robingenz Thanks for your review and the help. But are you sure you published that version? It does not seem to exist and does not show up on npmjs.com either.

**Comment by Crylion** [comment]
@robingenz No worries, we now published our dev version to our private registry. So we are set for now and can just wait for the official release :)

**Comment by robingenz** [comment]
Okay, i just published the dev version to the public npm registry.

File: .changeset/loud-dancers-look.md
Comment (robingenz) [inline .changeset/loud-dancers-look.md]:
```suggestion
feat(android): support spinner mode
```

**Comment by robingenz** [comment]
@c-clasing @Crylion I just tested it and it works well! Thank you for your contribution!

**Pull request merged**
